woocommerce/woocommerce
# Include product name as alt text for product thumbnail

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #51899.

This PR adds alt text to product thumbnails in the `woocommerce_cart` shortcode and Classic Cart block.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Ensure you have at least one product with a thumbnail in your store.
2. Add this product to the cart.
3. Create a new page and add the Classic Cart block(``)
4. Open the page on the frontend and inspect the source code for the product section.
5. Verify that the product thumbnail includes an alt text matching the product name.
6. Next, create a new page (or modify an existing one) and insert the `woocommerce_cart` shortcode(`[woocommerce_cart]`)
7. Repeat steps 4 and 5.

Expected outcome: The product thumbnail should display an alt text that corresponds to the product name:
<img width="757" alt="image" src="https://github.com/user-attachments/assets/cd1628bb-dfa6-4997-b682-1b82bf339270">




### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

Set the product name as alt text for the main product thumbnail when no alt text is provided.

</details>
